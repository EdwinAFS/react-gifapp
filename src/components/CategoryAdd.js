import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({setCategories}) => {

	const [categoryInput, setCategoryInput] = useState('');

	const handleChangeCategory = (e) => {
		setCategoryInput(e.target.value);
	}

	const handleSubmit = ( e ) => {
		e.preventDefault();

		if( categoryInput.trim().length < 2){
			return;
		}

		setCategories( categories => [categoryInput, ...categories] );
	}

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				value={categoryInput}
				onChange={handleChangeCategory}
			/>
		</form>
	);
};

CategoryAdd.propTypes = {
	setCategories: PropTypes.func.isRequired
}