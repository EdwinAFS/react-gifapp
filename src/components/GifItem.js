import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {

	return (
		<div className='card animate__animated animate__fadeInDown'>
			<div className='body'>
				<img src={url} alt={title}/>
			</div>
			<div className='footer'>
				<p>{title}</p>
			</div>
		</div>
	);
}


GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired
}