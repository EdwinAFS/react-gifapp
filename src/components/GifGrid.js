import React, { Fragment } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem'; 
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

	const { data: gifs, loading } = useFetchGifs( category );

	return (
		<Fragment>
			<h3>{category}</h3>

			{ loading && <p className='animate__animated animate__lightSpeedInLeft'>Cargando...</p>}
			<div className='grid'>
				{
					gifs.map(gif => {
						return (
							<GifItem key={gif.id} {...gif} />
						);
					})
				}
			</div>

		</Fragment>
	);

}

GifGrid.protoType = {
	category: PropTypes.string.isRequired
}