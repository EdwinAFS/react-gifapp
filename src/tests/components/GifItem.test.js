import { shallow } from "enzyme";
import { useEffect } from "react";
import { GifItem } from "../../components/GifItem";

describe('Test para el componente GifItem.js', ()=>{

	const title = 'Test';
	const url = 'https://media.giphy.com/media/3o6wrk0rO35GuXFP6o/giphy.gif';

	const wrapper = shallow(<GifItem 
		title={title} 
		url={url} 
	/>);

	test('Deberia mostrar el componente de GifItem', ()=>{	
		expect( wrapper ).toMatchSnapshot();
	});

	test('Deberia tener un parrafo con el titulo', () => {
		const titleValue = wrapper.find('p').text().trim(); 
		expect( titleValue ).toBe( title );
	});
	
	test('Deberia tener url y alt de los props', () => {
		const img = wrapper.find('img'); 
		expect( img.prop('src') ).toBe( url );
		expect( img.prop('alt') ).toBe( title );
	});
	
	test('Deberia contener la clase animate__fadeInDown', () => {
		const card = wrapper.find('.card'); 
		expect( card.prop('className') ).toContain( 'animate__fadeInDown' );
	});

});