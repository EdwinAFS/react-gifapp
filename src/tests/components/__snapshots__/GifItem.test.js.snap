// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente GifItem.js Deberia mostrar el componente de GifItem 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <div
    className="body"
  >
    <img
      alt="Test"
      src="https://media.giphy.com/media/3o6wrk0rO35GuXFP6o/giphy.gif"
    />
  </div>
  <div
    className="footer"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
